test/unit/math/rev/mat/functor/integrate_ode_adams_prim_test.cpp

```diff
@@ -356,10 +356,9 @@ TEST(StanMathOde_integrate_ode_adams, error_conditions_bad_ode) {
   std::vector<int> x_int(2, 0);
 
   std::string error_msg
-      = "ode_system: size of state vector y (2) and derivative vector dy_dt "
-        "(3) "
-        "in the ODE functor do not match in size.";
+      = "cvodes_ode_data: dz_dt (3) and states (2) must match in size";
```

Comment:
I think the change in error type is good. I think we should revisit the error message. Is this clear? This will be pushed all the way to the user and I think the older message is clearer.

Reply:
This error message is aligned with the non-stiff system. I agree that the old error message is easier to understand, but then we would need to change the non-stiff system as well and in addition we cannot use anymore our check functions here. I suggest to either (a) leave it in this pull for now as is and we make another issue+pull for improving error messages in both systems or (b) we need to touch the non-stiff system as well. I would prefer a.